packages/facility-server/app/sync/FacilitySyncManager.js

```diff
@@ -194,15 +197,18 @@ export class FacilitySyncManager {
     log.info('FacilitySyncManager.snapshottingOutgoingChanges', { pushSince });
     const modelsForPush = getModelsForPush(this.models);
     const outgoingChanges = await snapshotOutgoingChanges(this.sequelize, modelsForPush, pushSince);
-    if (outgoingChanges.length > 0) {
+    const outgoingChangesWithChangelogs = await attachChangelogToSnapshotRecords(this.sequelize, outgoingChanges, {
```

Comment:
Could be lower. Is it needed for that test spy? If not, would put it right above `pushOutgoingChanges`. Otherwise, at least within the .length check